OpenRA.Mods.Common/Effects/Beacon.cs

```diff
@@ -48,17 +49,28 @@ public class Beacon : IEffect, IScriptBindable, IEffectAboveShroud
 			this.duration = duration;
 			this.delay = delay;
 
+			var world = owner.World;
+			var bounds = new Rectangle();
+
 			if (!string.IsNullOrEmpty(arrowSprite))
 			{
-				arrow = new Animation(owner.World, beaconCollection);
+				arrow = new Animation(world, beaconCollection);
 				arrow.Play(arrowSprite);
+				bounds = Rectangle.Union(bounds, arrow.Image.Bounds);
```

Comment:
I believe that the default Rectangle has a position of 0, so unioning this with that will give a giant box that extends to the corner of the map.